core-bundle/src/Security/TwoFactor/TrustedDeviceManager.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Security\TwoFactor;
+
+use Contao\CoreBundle\Entity\TrustedDevice;
+use Contao\CoreBundle\Repository\TrustedDeviceRepository;
+use Contao\User;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\NonUniqueResultException;
+use Scheb\TwoFactorBundle\Model\TrustedDeviceInterface;
+use Scheb\TwoFactorBundle\Security\TwoFactor\Trusted\TrustedDeviceManagerInterface;
+use Scheb\TwoFactorBundle\Security\TwoFactor\Trusted\TrustedDeviceTokenStorage;
+use Symfony\Component\HttpFoundation\RequestStack;
+use UAParser\Parser;
+
+class TrustedDeviceManager implements TrustedDeviceManagerInterface
+{
+    private const DEFAULT_TOKEN_VERSION = 0;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var TrustedDeviceTokenStorage
+     */
+    private $trustedTokenStorage;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    public function __construct(RequestStack $requestStack, TrustedDeviceTokenStorage $trustedTokenStorage, EntityManagerInterface $entityManager)
+    {
+        $this->requestStack = $requestStack;
+        $this->trustedTokenStorage = $trustedTokenStorage;
+        $this->entityManager = $entityManager;
+    }
+
+    public function addTrustedDevice($user, string $firewallName): void
+    {
+        if (!$user instanceof User || !$user instanceof TrustedDeviceInterface) {
+            return;
+        }
+
+        /** @var TrustedDeviceRepository $trustedDeviceRepository */
+        $trustedDeviceRepository = $this->entityManager->getRepository(TrustedDevice::class);
```

Comment:
As the entity repository only has two methods, which are only used by the `TrustedDeviceManager` (which kinda is the only service supposed to fiddle with trusted devices), I wonder if we should remove the repository and just build the queries directly in this class?

Reply:
Fixed in 1dbb8d9.